app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -79,6 +79,32 @@ def metrics_connect
                                      authentication_password('default'))
     end
 
+    def run_generic_operation(operation, path)
```

Comment:
I don't like exposing `generic_operation` on the Middleware Manager API like this.  It seems like the Middleware Manager API should wrap these in something meaningful.  Like having an actual `reload_middleware_server` method.

Exposing this makes the caller have to know what the underlying Hawkular library requires in order to run various generic operations.

What about making `run_generic_operation` private and having a public `reload_middleware_server(server)` method here.